magnet_derive/src/meta.rs

```diff
@@ -4,6 +4,21 @@ use std::f64;
 use syn::{ Attribute, Meta, NestedMeta, MetaNameValue, Lit };
 use error::{ Error, Result };
 
+pub fn doc_meta(attrs: &[Attribute]) -> Option<MetaNameValue> {
+    let mut value = attrs.iter().filter_map(|attr| {
+        if let Some(meta) = attr.interpret_meta() {
```

Comment:
This can be simplified to:

```rust
attr.interpret_meta().and_then(|meta| match meta {
    Meta::NameValue(ref val) if val.ident == "doc" => Some(val.clone()),
    _ => None,
})
```